wave_geometry/include/wave/geometry/numerical_test_functors.hpp

```diff
@@ -40,13 +40,104 @@ class RotateAndJacobianJparamFunctor {
     }
 };
 
+class ComposeAndJacobianJLeftFunctor {
+ public:
+    Rotation R_left;
+    Rotation R_right;
+    ComposeAndJacobianJLeftFunctor(Rotation R_left, Rotation R_right) {
+        this->R_left = R_left;
+        this->R_right = R_right;
+    }
 
-// Implement the numerical differentiation, following procedure
-// for a 5 point stencil https://en.wikipedia.org/wiki/Five-point_stencil
-// Exact step size representation is outlined in
-// "Numerical Recipes by Brian P. Flannery,
-// Saul Teukolsky, and William H. Press, chapter 5.7.
+    Rotation operator()(Vec3 perturbation) {
+        Mat3 J;
+        Rotation R_perturbed = this->R_left;
+        R_perturbed = R_perturbed.manifoldPlus(perturbation);
+        return R_perturbed.composeAndJacobian(R_right, J, J);
+    }
+};
 
+class ComposeAndJacobianJRightFunctor {
+ public:
+    Rotation R_left;
+    Rotation R_right;
+    ComposeAndJacobianJRightFunctor(Rotation R_left, Rotation R_right) {
+        this->R_left = R_left;
+        this->R_right = R_right;
+    }
+
+    Rotation operator()(Vec3 perturbation) {
+        Mat3 J;
+        Rotation R_perturbed = this->R_right;
+        R_perturbed = R_perturbed.manifoldPlus(perturbation);
+        return R_left.composeAndJacobian(R_perturbed, J, J);
+    }
+};
+
+class InverseAndJacobianFunctor {
+ public:
+    Rotation R;
+    InverseAndJacobianFunctor(Rotation R) {
+        this->R = R;
+    }
+
+    Rotation operator()(Vec3 perturbation) {
+        Mat3 J;
+        Rotation R_perturbed = this->R;
+        R_perturbed = R_perturbed.manifoldPlus(perturbation);
+        return R_perturbed.inverseAndJacobian(J);
+    }
+};
+
+class LogMapAndJacobianFunctor {
+ public:
+    Rotation R;
+    LogMapAndJacobianFunctor(Rotation R) {
+        this->R = R;
+    }
+
+    Vec3 operator()(Vec3 perturbation) {
+        Mat3 J;
+        Rotation R_perturbed = this->R;
+        R_perturbed = R_perturbed.manifoldPlus(perturbation);
+        return Rotation::logMapAndJacobian(R_perturbed, J);
+    }
+};
+
+
+class ManifoldMinusAndJacobianJLeftFunctor {
+ public:
+    Rotation R_left;
+    Rotation R_right;
+    ManifoldMinusAndJacobianJLeftFunctor(Rotation R_left, Rotation R_right) {
+        this->R_left = R_left;
+        this->R_right = R_right;
+    }
+
+    Vec3 operator()(Vec3 perturbation) {
+        Mat3 J;
```

Comment:
it's a "dummy" variable that is passed into composeAndJacobian(R_right, J, J);

We don't use J, only the return value of composeAndJacobian.  In a future PR, we should check for null arguments on the jacobians and only compute them if asked for